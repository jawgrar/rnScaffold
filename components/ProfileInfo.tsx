import React, {FC, memo} from 'react';
import {Text, View} from 'react-native';

interface ProfileInfoProps {
  displayName: string | null;
  email: string | null;
}

const ProfileInfo: FC<ProfileInfoProps> = ({displayName, email}) => (
  <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
    <Text>{displayName || 'unknown'}</Text>
    {email && <Text style={{fontSize: 10, marginVertical: 8}}>{email}</Text>}
  </View>
);

export default memo(ProfileInfo);
