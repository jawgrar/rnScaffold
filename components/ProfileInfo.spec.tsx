import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProfileInfo from './ProfileInfo';

test('renders correctly', () => {
  const tree = renderer
    .create(<ProfileInfo displayName="Lorem Ipsum" email="user@test.com" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
