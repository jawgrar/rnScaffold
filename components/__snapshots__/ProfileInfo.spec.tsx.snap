// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Lorem Ipsum
  </Text>
  <Text
    style={
      Object {
        "fontSize": 10,
        "marginVertical": 8,
      }
    }
  >
    user@test.com
  </Text>
</View>
`;
